cybergis_compute_client/CyberGISCompute.py

```diff
@@ -67,7 +68,63 @@ def __init__(self, url="cgjobsup.cigi.illinois.edu", port=443, protocol='HTTPS',
         self.job = None
         self.recentDownloadPath = None
 
-    def login(self, manualLogin=True, verbose=True):
+    def encrypt_token(self, token):
+        self.jupyterhubApiToken = base64.b64encode((self.jupyterhubHost + '@' + token).encode('ascii')).decode('utf-8')
+
+    def get_jupyterhubHost(self):
+        if self.jupyterhubHost is None:
+            print("Please copy the JupyterHub url along with port. E.g http://127.0.0.1:8081")
+            self.jupyterhubHost = input('Enter your jupyterhubHost here: ')
+
+    def set_username(self):
+        res = self.client.request('GET', '/user', {"jupyterhubApiToken": self.jupyterhubApiToken})
+        self.username = res['username']
+
+    def save_token(self):
+        with open('./cybergis_compute_user.json', 'w') as json_file:
+            json.dump({"token": self.jupyterhubApiToken}, json_file)
+
+    def login_token(self):
+        try:
+            self.set_username()
+            self.save_token()
+            return self.login()
+        except:
+            print('❌ Failed to login via system token')
+
+    def login_manual(self):
+        if self.isJupyter:
```

Comment:
Everything works on CyberGISX, but when I try to use this on my laptop, it isn't too much of an improvement over what I used to have to do because it isn't asking me for the hostname. 

`cybergis.jupyterhubHost` is set to `localhost:8888` when I instantiate the CyberGISCompute object and it never asks me to specify a host since it's not "None". I think the best quick fix would be to add a "hostname"/"authenticator" argument to the "login" function which allows the user to specify where they want to authenticate against. Let me know if that makes sense.

Reply:
Done ! @alexandermichels Can you have a look now ?